policy/modules/system/systemd.if

```diff
@@ -58,28 +69,603 @@ template(`systemd_role_template',`
+	dbus_manage_session_dbus_runtime_sock_files($1_systemd_t)
 	dbus_system_bus_client($1_systemd_t)
 
+	dev_read_urand($1_systemd_t)
+
+	fs_manage_cgroup_files($1_systemd_t)
+	fs_watch_cgroup_files($1_systemd_t)
+
+	init_search_units($1_systemd_t)
+	init_read_generic_units_files($1_systemd_t)
+
+	kernel_dontaudit_getattr_proc($1_systemd_t)
+
+	manage_dirs_pattern($1_systemd_t, systemd_user_home_t, systemd_user_home_t)
+	manage_files_pattern($1_systemd_t, systemd_user_home_t, systemd_user_home_t)
+	manage_lnk_files_pattern($1_systemd_t, systemd_user_home_t, systemd_user_home_t)
+
+	manage_dirs_pattern($1_systemd_t, systemd_user_home_unit_t, systemd_user_home_unit_t)
+	manage_files_pattern($1_systemd_t, systemd_user_home_unit_t, systemd_user_home_unit_t)
+	manage_lnk_files_pattern($1_systemd_t, systemd_user_home_unit_t, systemd_user_home_unit_t)
+
+	manage_dirs_pattern($1_systemd_t, systemd_user_runtime_t, systemd_user_runtime_t)
+	manage_files_pattern($1_systemd_t, systemd_user_runtime_t, systemd_user_runtime_t)
+	manage_lnk_files_pattern($1_systemd_t, systemd_user_runtime_t, systemd_user_runtime_t)
+	manage_sock_files_pattern($1_systemd_t, systemd_user_runtime_t, systemd_user_runtime_t)
+	manage_fifo_files_pattern($1_systemd_t, systemd_user_runtime_t, systemd_user_runtime_t)
+	manage_chr_files_pattern($1_systemd_t, systemd_user_runtime_t, systemd_user_runtime_t)
+
 	selinux_use_status_page($1_systemd_t)
 
 	seutil_read_file_contexts($1_systemd_t)
 	seutil_search_default_contexts($1_systemd_t)
+
+	systemd_list_journal_dirs($3)
+	systemd_read_journal_files($3)
+
+	init_list_unit_dirs($1_systemd_t)
+
+	systemd_manage_user_home_content($3)
+	systemd_relabel_user_home_content($3)
+
+	systemd_manage_user_home_unit_files($3)
+	systemd_relabel_user_home_unit_files($3)
+
+	systemd_search_user_runtime_unit_dirs($1_systemd_t)
+	manage_dirs_pattern($1_systemd_t, systemd_user_runtime_unit_t, systemd_user_runtime_unit_t)
+	manage_files_pattern($1_systemd_t, systemd_user_runtime_unit_t, systemd_user_runtime_unit_t)
+	manage_lnk_files_pattern($1_systemd_t, systemd_user_runtime_unit_t, systemd_user_runtime_unit_t)
+
+	systemd_read_user_unit_files($3)
+	systemd_read_user_unit_files($1_systemd_t)
+
+	systemd_search_user_runtime_unit_dirs($1_systemd_t)
+	systemd_list_user_runtime_unit_dirs($3)
+	systemd_read_user_runtime_units($3)
+
+	systemd_status_user_units($3)
+	systemd_start_user_units($3)
+	systemd_stop_user_units($3)
+	systemd_reload_user_units($3)
+	systemd_status_user_home_units($3)
+	systemd_start_user_home_units($3)
+	systemd_stop_user_home_units($3)
+	systemd_reload_user_home_units($3)
+
+	type_transition systemd_user_session_type systemd_user_runtime_t:dir systemd_user_runtime_unit_t "generator.early";
+	type_transition systemd_user_session_type systemd_user_runtime_t:dir systemd_user_runtime_unit_t "generator.late";
+	type_transition systemd_user_session_type systemd_user_runtime_t:dir systemd_user_runtime_unit_t "transient";
+
```

Comment:
you mention /run/user/UID/systemd/user in the fc specs above but its not included here. I don't think /run/user/UID/systemd/user is used in the first place but regardless: if you do it then do it consistently

Reply:
Adding the missing type transition here per above.